modules/preview-react/side-panel/lib/SidePanel.tsx

```diff
@@ -192,70 +140,44 @@ export const SidePanel = createComponent('section')({
       }
     }, [state, onStateTransition]);
 
-    const handleAnimationEnd = (event: React.AnimationEvent<HTMLDivElement>) => {
-      if (event.currentTarget === event.target) {
-        if (event.animationName === sidepanelCollapsedKeyFrames) {
-          setState('collapsed');
-        }
-
-        if (event.animationName === sidepanelExpandedlKeyFrames) {
+    const handleAnimationEnd = (event: React.TransitionEvent<HTMLDivElement>) => {
+      if (event.propertyName === 'width') {
+        if (state === 'expanding') {
           setState('expanded');
+        } else if (state === 'collapsing') {
+          setState('collapsed');
         }
       }
-
-      if (typeof onAnimationEnd !== 'undefined') {
-        onAnimationEnd(event);
-      }
     };
 
-    const handleAnimationStart = (event: React.AnimationEvent<HTMLDivElement>) => {
-      if (event.currentTarget === event.target) {
-        if (
-          event.animationName === sidepanelCollapsedKeyFrames ||
-          event.animationName === sidepanelExpandedlKeyFrames
-        ) {
-          setState(expanded ? 'expanding' : 'collapsing');
-        }
-      }
-
-      if (typeof onAnimationStart !== 'undefined') {
-        onAnimationStart(event);
+    const handleAnimationStart = () => {
+      if (state === 'collapsed' || state === 'collapsing') {
+        setState('expanding');
+      } else if (state === 'expanded' || state === 'expanding') {
+        setState('collapsing');
       }
```

Comment:
Same here:
`setState(state.replace(/ed$/, 'ing'));`

Reply:
Al though this is less code, I find it less clear to read. I prefer the explicit logic so that whomever jumps into this code again can read through the logic.

Reply:
That's a little too clever. I'd be okay with a little less syntax, but still clear in what's going on:

```tsx
      if (['collapsed', 'collapsing'].includes(state)) {
        setState('expanding');
      } else if (['expanded', 'expanding'].includes(state)) {
        setState('collapsing');
      }
```